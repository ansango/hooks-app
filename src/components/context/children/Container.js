
import "./Container.css";
export const Container = ({ title = "", children }) => {
  return (
    <div className="base-container py-5 bg-light">
      <h3 className="text-center">{title}</h3>
      <div className="px-2">{children}</div>
    </div>
  );
};
