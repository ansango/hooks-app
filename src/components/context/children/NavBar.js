import { NavLink } from "react-router-dom";

export const NavBar = ({ routes }) => {
  return (
    <nav className="navbar navbar-light bg-light d-flex flex-row justify-content-end">
      <div className="navbar-nav">
        <div className="d-flex">
          {routes.map((route) => (
            <NavLink
              key={route.name}
              exact
              activeClassName="active"
              className="nav-item nav-link mx-3"
              to={route.path}
            >
              {route.name}
            </NavLink>
          ))}
        </div>
      </div>
    </nav>
  );
};
