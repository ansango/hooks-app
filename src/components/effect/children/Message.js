import { useEffect, useState } from "react";

export const Message = () => {
  const [coords, setCoords] = useState({ x: 0, y: 0 });
  const { x, y } = coords;
  useEffect(() => {
    const mouseMove = (e) => setCoords({ x: e.x, y: e.y });
    window.addEventListener("mousemove", mouseMove);
    return () => {
      window.removeEventListener("mousemove", mouseMove);
    };
  }, []);
  return (
    <div>
      X: <p>{x}</p>
      Y: <p>{y}</p>
    </div>
  );
};
