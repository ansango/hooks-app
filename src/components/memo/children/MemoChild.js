import React from "react";

export const MemoChild = React.memo(({ number, increment }) => {
  console.log("render");

  return (
    <button className="btn btn-primary me-3" onClick={() => increment(number)}>
      {number}
    </button>
  );
});
