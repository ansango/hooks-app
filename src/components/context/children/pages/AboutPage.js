import { Container } from "../Container";

export const AboutPage = () => {
    return <Container title="About"></Container>;
};
