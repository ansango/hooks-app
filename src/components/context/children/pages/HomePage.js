import { useContext } from "react";
import { log } from "../../../../helpers/log";
import { Container } from "../Container";
import { UserContext } from "../UserContext";

export const HomePage = () => {
  const userContext = useContext(UserContext);
  log("MainApp - HomePage",userContext)
  return <Container title="Home"></Container>;
};
