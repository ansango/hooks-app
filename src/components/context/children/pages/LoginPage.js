import { Container } from "../Container";

export const LoginPage = () => {
  return <Container title="Login"></Container>;
};
