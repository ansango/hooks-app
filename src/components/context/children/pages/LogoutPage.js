import { useContext } from "react";
import { log } from "../../../../helpers/log";
import { Container } from "../Container";
import { UserContext } from "../UserContext";

export const LogoutPage = () => {
  const { user, setUser } = useContext(UserContext);
  log("MainApp - LogoutPage", { user, setUser });

  const handleClick = () => setUser({});
  return (
    <Container title="Logout">
      <pre>{JSON.stringify(user, null, 3)}</pre>
      <button className="btn btn-warning" onClick={handleClick}>
        Logout
      </button>
    </Container>
  );
};
