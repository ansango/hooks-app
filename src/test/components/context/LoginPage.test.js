import { mount } from "enzyme";
//import { useHistory } from "react-router-dom";
import { LoginPage } from "../../../components/context/children/pages/LoginPage";
import { UserContext } from "../../../components/context/children/UserContext";

jest.mock("react-router-dom", () => ({
  useHistory: () => ({ push: jest.fn() }),
}));

describe("Test in <LoginPage/>", () => {
  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginPage />
    </UserContext.Provider>
  );

  test("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should set user", () => {
    wrapper.find("button").prop("onClick")();
    expect(setUser).toHaveBeenCalledWith({ id: 123, name: "Anibal" });
  });
});
