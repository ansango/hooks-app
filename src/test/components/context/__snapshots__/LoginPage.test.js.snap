// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LoginPage/> should render 1`] = `
<div
  className="base-container py-5 bg-light"
>
  <h3
    className="text-center"
  >
    Login
  </h3>
  <div
    className="d-flex flex-column align-items-center justify-content-center p-5"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
